File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/POLIMORFISMO/Ejercicio3py_java/Matriz.java
package Ejercicio3py_java;

public class Matriz {
    private final int FILAS = 10;
    private final int COLUMNAS = 10;
    private float[][] matriz;

    public Matriz() {
        matriz = new float[FILAS][COLUMNAS];
        for (int i = 0; i < FILAS; i++) {
            matriz[i][i] = 1.0f;
        }
    }

    public Matriz(float[][] datos) {
        if (datos.length != FILAS || datos[0].length != COLUMNAS) {
            throw new IllegalArgumentException("La matriz debe ser de 10x10");
        }

        matriz = new float[FILAS][COLUMNAS];
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                matriz[i][j] = datos[i][j];
            }
        }
    }

    public void mostrar() {
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                System.out.print(matriz[i][j] + "\t");
            }
            System.out.println();
        }
    }

    public Matriz sumar(Matriz otra) {
        float[][] resultado = new float[FILAS][COLUMNAS];
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                resultado[i][j] = this.matriz[i][j] + otra.matriz[i][j];
            }
        }
        return new Matriz(resultado);
    }

    public Matriz restar(Matriz otra) {
        float[][] resultado = new float[FILAS][COLUMNAS];
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                resultado[i][j] = this.matriz[i][j] - otra.matriz[i][j];
            }
        }
        return new Matriz(resultado);
    }

    public boolean igual(Matriz otra) {
        for (int i = 0; i < FILAS; i++) {
            for (int j = 0; j < COLUMNAS; j++) {
                if (this.matriz[i][j] != otra.matriz[i][j]) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/POLIMORFISMO/Ejercicio3py_java/Matriz.py
class Matriz:
    def __init__(self, datos=None):
        self.filas = 10
        self.columnas = 10

        if datos is None:
            self.matriz = [[1.0 if i == j else 0.0 for j in range(self.columnas)] for i in range(self.filas)]
        else:
            if len(datos) != self.filas or any(len(fila) != self.columnas for fila in datos):
                raise ValueError("La matriz debe ser de 10x10")
            self.matriz = [[float(valor) for valor in fila] for fila in datos]

    def mostrar(self):
        for fila in self.matriz:
            print(fila)
    # SE DEBN SOBRECARAGA +, -  Y ==
    def sumar(self, otra):
        resultado = [[self.matriz[i][j] + otra.matriz[i][j] for j in range(self.columnas)] for i in range(self.filas)]
        return Matriz(resultado)

    def restar(self, otra):
        resultado = [[self.matriz[i][j] - otra.matriz[i][j] for j in range(self.columnas)] for i in range(self.filas)]
        return Matriz(resultado)

    def igual(self, otra):
        for i in range(self.filas):
            for j in range(self.columnas):
                if self.matriz[i][j] != otra.matriz[i][j]:
                    return False
        return True
